src/Pickles/Pickles/CommandLineArgumentParser.cs

```diff
@@ -122,6 +127,13 @@ public bool Parse(string[] args, IConfiguration configuration, TextWriter stdout
                 configuration.AddTestResultFiles(files.Select(f => this.fileSystem.FileInfo.FromFileName(f)));
             }
 
+            if (!string.IsNullOrEmpty(this.testResultsDirectory))
+            {
+                configuration.AddTestResultFiles(
+                    this.fileSystem.DirectoryInfo.FromDirectoryName(
+                        this.testResultsDirectory).EnumerateFiles("*.json", System.IO.SearchOption.AllDirectories));
```

Comment:
This restricts the results directory argument to .json result files.

I think it would be better to use the link-results-file parameter and make it able to read a wildcard expression.

For example, the link-results-file parameter should be able to interpret `c:\temp\tests\*.trx`.